gap/semigroups/semifp.gi

```diff
@@ -636,3 +636,163 @@ function(M)
 
   return MagmaIsomorphismByFunctionsNC(M, MF, map, inv);
 end);
+
+InstallMethod(ParseRelations,
+"for a list of free generators and a string",
+[IsDenseList, IsString],
+function(gens, inputstring)
+    local newinputstring, g, chartoel, RemoveBrackets, ParseRelation, output;
+
+    for g in gens do
+      if not (Size(String(g)) = 1 and String(g)[1]
+         in "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ") then
+        ErrorNoReturn(Concatenation(
+        "expected a single english letter but found ", String(g)));
```

Comment:
E

Reply:
Also this error message should indicate where the invalid thing occurred (e.g. something like "expected first argument to be a list of single English letters, but found ...")